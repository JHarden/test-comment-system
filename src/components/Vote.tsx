import React from "react";
import styled from "styled-components";

export interface VotesProps {
    isLoggedIn: boolean;
    votes: number;
    onVote: (votes: number) => void;
    hasVoted: boolean;
};

const BlueArrow = styled.div`
    width: 0;
    height: 0;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-bottom: 20px solid #4ca4ff;
`;

const GreyArrow = styled.div`
    width: 0;
    height: 0;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-bottom: 20px solid grey;
`;

const VoteWrapper = styled.div`
    width: 70px;
    height: 70px;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 65px;
    align-items: center;
    >div {
        padding-top: 10px;
        font-weight: bold;
    }
`;

class Vote extends React.Component<VotesProps, {}> {
    private handleOnClick = () => this.props.isLoggedIn && !this.props.hasVoted && this.props.onVote(this.props.votes + 1);
    
    render() {
        const { votes, hasVoted } = this.props;
        return (
            <VoteWrapper>
                {hasVoted ?
                    <BlueArrow onClick={this.handleOnClick}></BlueArrow>
                    :
                    <GreyArrow onClick={this.handleOnClick}></GreyArrow>
                }
                <div style={hasVoted? {color: '#4ca4ff'} : {color: 'grey', }}>{votes}</div>
            </VoteWrapper>
        )
    }
}

export default Vote;