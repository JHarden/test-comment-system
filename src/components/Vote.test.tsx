import React from 'react';
import Vote from './Vote';
import renderer from 'react-test-renderer'
const mock = () => { }

const props = {
  votes: 12,
  onVote: () => {},
  hasVoted: false
}

test('Vote component test', () => {
  const component = renderer.create(
    <Vote {...props}></Vote>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
