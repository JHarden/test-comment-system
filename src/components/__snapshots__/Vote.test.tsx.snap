// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent component test 1`] = `
<div>
  <div>
    <button
      onClick={[Function]}
    >
      Create Article
    </button>
  </div>
</div>
`;

exports[`Vote component test 1`] = `
<div
  style={
    Object {
      "color": "purple",
      "outline": "1px solid purple",
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "background": "red",
        "height": "30px",
        "width": "30px",
      }
    }
  >
     VOTE 
  </div>
  <div>
    12
  </div>
</div>
`;
